cg/services/run_devices/pacbio/run_validator/pacbio_run_validator.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+
+from cg.constants.constants import FileFormat
+from cg.services.decompression_service.decompressor import Decompressor
+from cg.services.run_devices.abstract_classes import RunValidator
+from cg.services.run_devices.pacbio.run_data_generator.run_data import PacBioRunData
+from cg.services.run_devices.pacbio.run_file_manager.models import PacBioRunValidatorFiles
+from cg.services.run_devices.pacbio.run_file_manager.run_file_manager import PacBioRunFileManager
+from cg.services.validate_file_transfer_service.validate_file_transfer_service import (
+    ValidateFileTransferService,
+)
+
+
+class PacBioRunValidator(RunValidator):
+    """PacBio run validator.
+    Ensure that the run can start post-processing of a pacbio run can start.
+    """
+
+    def __init__(
+        self,
+        decompressor: Decompressor,
+        file_transfer_validator: ValidateFileTransferService,
+        file_manager: PacBioRunFileManager,
+    ):
+        self.decompressor = decompressor
+        self.file_transfer_validator = file_transfer_validator
+        self.file_manager = file_manager
+
+    def ensure_post_processing_can_start(self, run_data: PacBioRunData):
+        """Ensure that a post processing run can start."""
+        paths_information: PacBioRunValidatorFiles = self.file_manager.get_run_validation_files(
+            run_data
+        )
+        self.file_transfer_validator.validate_file_transfer(
+            manifest_file=paths_information.manifest_file,
+            source_dir=run_data.full_path,
+            manifest_file_format=FileFormat.TXT,
+        )
+        self.decompressor.decompress(
```

Comment:
Is this something we always want to do? For me it would make sense to have a "if needs decompressing" or something?

Reply:
Might be that we only start the decompression here though. If so, feel free to resolve.

Reply:
We always want to do that as the metric reports come zipped. If already zipped I think it does not matter to overwrite the existing ones. If there is no zip file then an error should be raised

Reply:
The error should be raised in the validate_file_transfer though right?

Reply:
added the handling in the runfile manager